File: C/Curso 1/adivinhacao.c
#include <stdio.h> // o include é a maneira utilizada para importar bibliotecas
#include <stdlib.h>
#include <time.h>

// caso haja a necessidade de criar uma constante utiliza-se #define, o nome da constante é colocado tudo em maiúsculo

int numerosecreto;
int chute;
unsigned int acertou;
double pontos = 1000;
int tentativas;
int nivel;


int main(){
    printf("Bem vindo ao jogo de adivinhação!\n");
    
    printf("Qual o nível de dificuldade desejada?\n");
    printf("1: Fácil\n");
    printf("2: Médio\n");
    printf("3: Difícil\n");
    printf("Escolha:");
    scanf("%d", &nivel);
    printf("\n");

    switch (nivel) {
        case 1:
            tentativas = 20;
            break;
        case 2:
            tentativas = 15;
            break;

        default:
            tentativas = 6;
            break; 
    }

    int seed = time(0); // a função time() serve para retornar o número de segundos desde janeiro de 1970(EPOCH) e como esse tempo muda constantemente utilizaremos ele como seed para o número secreto
    srand(seed);

    numerosecreto = rand() % 100; // para o número secreto ser sempre entre 0 e 99, pegaremos o resto da divisão entre o número aleatório e o número 100

    while (tentativas> 0){
        printf("Tentativas: %d\n", tentativas);
    
        printf("Faça o seu chute: ");
        scanf("%d", &chute);
        printf("Seu chute foi: %d\n", chute);

        acertou = chute == numerosecreto;

        if (acertou){
            printf("Parabéns! Você acertou!\n");
            break;
        } else {
            printf("Você errou! Tente novamente!\n");
            pontos -= abs((double)((chute - numerosecreto)/2)); // esse (double) é utilizado para fazer o casting das variáveis inteiras para elas se comportarem como um double
            tentativas -= 1;

            printf("Dica: ");
            if (chute > numerosecreto){
                printf("Seu chute foi maior que o número secreto.\n");
            } else if (chute < numerosecreto) {
                printf("Seu chute foi menor que o número secreto.\n");
            }
        }
    }

    printf("Fim de jogo!\n");
    printf("O número secreto era: %d\n", numerosecreto);
    printf("Você fez %.1f pontos!\n", pontos);

    return 0;
}

File: PYTHON/Aula  Python 1/Adivinhacao.py
import random

def jogar() -> None:

    print('*********************************')
    print('Bem vindo ao jogo de adivinhação!')
    print('*********************************')

    numero_secreto = random.randrange(1, 101)
    tentativas = 20
    rodada = 1
    pontos = 1000

    print('Qual nível de dificuldade você quer?')
    print('''
    (1) Fácil
    (2) Médio
    (3) Dificil
    ''')
    dificuldade = input("Dificuldade:")

    if dificuldade == '1':
        tentativas = 20
    elif dificuldade == '2':
        tentativas = 10
    elif dificuldade == '3':
        tentativas = 5
    else:
        print("A dificuldade está na fácil")

    total_de_rodadas = tentativas

    while tentativas > 0:
        print(f"RODADA {rodada} DE {total_de_rodadas}")
        chute = int(input("Digite seu número de 1 a 100:"))

        acertou = chute == numero_secreto
        menor = chute < numero_secreto
        maior = chute > numero_secreto

        print("você digitou ",chute)
        if not (1 < chute > 100):
            print("Digite um número de 1 a 100")

        if acertou:
            print("Você acertou!")
            print(f"O seu total de pontos foi {pontos} pontos")
            break
        elif menor:
            print("Seu chute foi menor que o número")
        elif maior:
            print("Seu chute foi maior que o número")
        
        pontos_perdidos = abs(numero_secreto - chute)
        pontos = pontos - pontos_perdidos
        tentativas -= 1
        rodada += 1
        
        print()
        print(f"você tem {tentativas} tentativas")
        print()
        if tentativas == 0:
            print("Você perdeu!")
            print(f"A sua pontuação foi {pontos} pontos")

    print('Fim de jogo!')

if __name__ == "__main__":
    jogar()